server/script/migrateAssetsLocation.js

```diff
@@ -0,0 +1,252 @@
+  const data = await getNewMeta(fileMetas, meta, id);
+  return { data };
+}
+
+async function migrateRepositoryActivities(repositoryId, transaction) {
+  const activities = await Activity.findAll(
+    { where: { repositoryId } },
+    { transaction }
+  );
+  return Promise.each(activities, async it => {
+    const payload = await migrateActivity(it);
+    return it.update(payload, { transaction });
+  });
+}
+
+async function migrateRepositoryContentElements(repositoryId, transaction) {
+  const contentElements = await ContentElement.findAll(
+    { where: { repositoryId } },
+    { transaction }
+  );
+  return Promise.each(contentElements, async it => {
+    const payload = await migrateContentElement(it);
+    return it.update(payload, { transaction });
+  });
+}
+
+async function migrateRepositoryRevisions(repositoryId, transaction) {
+  const options = {
+    where: { repositoryId, entity: { [Op.in]: REVISION_TYPES } },
+    transaction
+  };
+  const count = await Revision.count(options);
+  const pages = Math.ceil(count / CHUNK_SIZE);
+  return Promise.each(
+    Array.from({ length: pages }, (_, i) => i + 1),
+    page => migrateRevisionsChunk({ page, options, transaction })
+  );
+}
+
+async function migrateRevisionsChunk({ page, options, transaction }) {
+  const offset = (page - 1) * CHUNK_SIZE;
+  const revisions = await Revision.findAll({
+    ...options,
+    offset,
+    limit: CHUNK_SIZE
+  });
+  return Promise.each(revisions, async it => {
+    const payload = await migrateRevision(it);
+    return it.update(payload, { transaction });
+  });
+}
+
+async function migrateActivity(activity) {
+  const { repositoryId, data: meta } = activity;
+  const fileMetas = getActivityMetadata(activity)
+    .filter(it => it.type === 'FILE')
+    .map(it => it.key);
+  const data = await getNewMeta(fileMetas, meta, repositoryId);
+  return { data };
+}
+
+async function migrateContentElement(element) {
+  const data = await migrateContentElementData(element);
+  const meta = await migrateContentElementMeta(element);
+  return { data, meta };
+}
+
+function migrateContentElementData(element) {
+  const { type, data } = element;
+  if (type === 'IMAGE') return imageMigrationHandler(element);
+  if (data.embeds) return embedsMigrationHandler(element);
+  if (data.assets) return defaultMigrationHandler(element);
+  return data;
+}
+
+async function migrateContentElementMeta(element) {
+  const { repositoryId } = element;
+  const metaInputs = schemasIds
+    .map(id => getElementMetadata(id, element))
+    .filter(meta => !meta.isEmpty)
+    .map(meta => meta.inputs);
+  const fileMetas = flatten(metaInputs)
+    .filter(it => it.type === 'FILE')
+    .map(it => it.key);
+  return getNewMeta(fileMetas, element.meta, repositoryId);
+}
+
+async function getNewMeta(fileMetas, meta, repositoryId) {
+  const newMeta = await Promise.map(toPairs(meta), async it => {
+    const [id, value] = it;
+    if (!fileMetas.includes(id)) return it;
```

Comment:
Why iterating per data set, wouldn't it be more intuitive for each defined (config) file meta to check if it exists/is set ? 🤔